src/libs/actions/SignInRedirect.js

```diff
@@ -26,7 +27,8 @@ function clearStorageAndRedirect(errorMessage) {
     // Clearing storage discards the authToken. This causes a redirect to the SignIn screen
     Onyx.clear()
         .then(() => {
-            if (preferredLocale) {
+            // Betas are only present when user is Logined
+            if (Permissions.canUseInternationalization([]) && preferredLocale) {
```

Comment:
Ah hmm not sure how `isDevelopment` relates here. I think if we are logging out it maybe makes sense to just clear the `preferredLocale`? Why are we trying to persist it? Maybe @iwiznia knows.

Reply:
If I have selected spanish locale and log out (or get logged out), then I want to see the login screen in spanish, not english. In any case it's fine to only check for the betas in the UI, if a user not on the beta sets the locale manually via developer console or something, then it's ok if the localization feature works for them,

Reply:
It's needed. Otherwise, the user will be locked in one language. If we are not showing a LocalePicker on the login page in staging then the language should not change to Spanish at all after logout, if the logged-in user was using spanish.

We determine whether the locale picker is shown or not, is via beta permissions check so I have used the same here.

Reply:
I think this check is confusing. Calling `canUseInternationalization` passing an empty betas list is just a confusing way to check `isDevelopment()` in this case? I think we should completely remove this check from here

Reply:
Hmm. It can be. I will test this way.

Reply:
>I think this check is confusing. Calling canUseInternationalization passing an empty betas list is just a confusing way to check isDevelopment() in this case

Agree